src/diffusers/models/unet_2d_condition.py

```diff
@@ -223,25 +259,29 @@ def __init__(
         # up
         reversed_block_out_channels = list(reversed(block_out_channels))
         reversed_attention_head_dim = list(reversed(attention_head_dim))
-        only_cross_attention = list(reversed(only_cross_attention))
+        reversed_attn1_types = list(reversed(attn1_types))
+        reversed_attn2_types = list(reversed(attn2_types))
+        
+        if isinstance(upsample, bool):
+            upsample = (upsample,) * len(up_block_types)
+
         output_channel = reversed_block_out_channels[0]
         for i, up_block_type in enumerate(up_block_types):
             is_final_block = i == len(block_out_channels) - 1
+            is_first_block = i == 0
 
             prev_output_channel = output_channel
             output_channel = reversed_block_out_channels[i]
             input_channel = reversed_block_out_channels[min(i + 1, len(block_out_channels) - 1)]
 
             # add upsample block for all BUT final layer
-            if not is_final_block:
-                add_upsample = True
-                self.num_upsamplers += 1
-            else:
-                add_upsample = False
+            add_upsample = not is_final_block if upsample is None else upsample[i]
+            if add_upsample:
+                 self.num_upsamplers += 1
 
             up_block = get_up_block(
                 up_block_type,
-                num_layers=layers_per_block + 1,
+                num_layers= layers_per_block + 1 if skip_freq == "layer" else layers_per_block,
```

Comment:
Instead of having a `skip_freq`, let's instead handle this directly in the new upsampling blocks